### PR TITLE
Proofread on *.md

### PR DESCRIPTION
**Pull request by ickc**
Proofreading *.md files (i.e. excluding `MANUAL.txt` for now). The commit messages should be self-explanatory. They mainly are:

- add relevant links
- proofread on grammar, syntax, etc.
- consistency

Feel free to squash this into 1 commit. I want them to be more atomic in case you want to cherry-pick.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -47,13 +47,13 @@ a bug in pandoc.  Quoting from the MANUAL:
 > to be perfect, conversions from formats more expressive than pandoc's
 > Markdown can be expected to be lossy.
 
-For example, both docx and odt can represent margin size, but because
-pandoc's internal document model does not contain a representation of
+For example, both `docx` and `odt` formats can represent margin size, but
+because pandoc's internal document model does not contain a representation of
 margin size, this information will be lost on converting from docx
-to odt.  (You can, however, customize margin size using `--reference-odt`.)
```
Comment (jgm) [inline CONTRIBUTING.md]: Actually this should now be `--reference-doc`!  Didn't notice tihs before.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -47,13 +47,13 @@ a bug in pandoc.  Quoting from the MANUAL:
...
+to `odt`.  (You can, however, customize margin size using `--reference-odt`.)
 
 So before submitting a bug report, consider whether it might be
-"out of scope."  If it concerns a feature of documents that isn't
```
Comment (jgm) [inline CONTRIBUTING.md]: This is actually the correct punctuation in American style (period inside quote).
It's illogical, I know.

File: doc/using-the-pandoc-api.md [hunk 3]
```diff
@@ -313,10 +313,10 @@ Note that this looks first in the
 users to override the system defaults.  If you want to disable
 this behavior, use `setUserDataDir Nothing`.
 
-To render a template, use `renderTemplate'`, which takes two
```
Comment (jgm) [inline doc/using-the-pandoc-api.md]: ACtually it is `renderTemplate'`, the prime is there in the function name.
(There's also `renderTemplate`, which is different.)

File: doc/using-the-pandoc-api.md [hunk 4]
```diff
@@ -313,10 +313,10 @@ Note that this looks first in the
...
+To render a template, use `renderTemplate`, which takes two
 arguments, a template (String) and a context (any instance
 of ToJSON).  If you want to create a context from the metadata
-part of a Pandoc document, use `metaToJSON'` from
```
Comment (jgm) [inline doc/using-the-pandoc-api.md]: It really is `metaToJSON'`.

**Comment by jgm** [comment]
Thanks!  I've made a few comments where changes are needed.

**Comment by ickc** [comment]
I added 3 commits to fix the things you commented (never knew the rules about that punctuation in American style, perhaps it is different in British English, that I was grown up with).

I added 3 more commits to different fixes, where the 0b09409 is about updating the years in the copyright. I checked the commit history of them to see they have been edited in the last 2 years. There are some cases like `src/Text/Pandoc/Readers/Haddock.hs` that I don't know how to update the copyright, because originally it was copyrighted by an author and later updated by others.

**Pull request merged**